sklearn/neighbors/tests/test_nca.py

```diff
@@ -0,0 +1,442 @@
+import pytest
+import re
+import numpy as np
+from numpy.testing import assert_array_equal, assert_array_almost_equal
+from scipy.optimize import check_grad
+from sklearn import clone
+from sklearn.exceptions import ConvergenceWarning
+from sklearn.utils import check_random_state
+from sklearn.utils.testing import (assert_raises, assert_equal,
+                                   assert_raise_message, assert_warns_message,
+                                   assert_true)
+from sklearn.datasets import load_iris, make_classification
+from sklearn.neighbors.nca import NeighborhoodComponentsAnalysis
+from sklearn.metrics import pairwise_distances
+
+
+rng = check_random_state(0)
+# load and shuffle iris dataset
+iris = load_iris()
+perm = rng.permutation(iris.target.size)
+iris_data = iris.data[perm]
+iris_target = iris.target[perm]
+EPS = np.finfo(float).eps
+
+
+def test_simple_example():
```

Comment:
It's hard to find such an example, because the cost function is not convex in general, but I'll commit an example where three points should collapse into one. Is it enough ?